source/BoardingPanel.h

```diff
@@ -123,10 +132,22 @@ class BoardingPanel : public Panel {
 	bool playerDied = false;
 	bool isCapturing = false;
 	bool isFirstCaptureAction = true;
+	bool isFirstCaptureAttempt = true;
+	bool isHijacking = false;
 	// Calculating the odds of combat success, and the expected casualties, is
 	// non-trivial. So, cache the results for all crew amounts up to full.
 	CaptureOdds attackOdds;
 	CaptureOdds defenseOdds;
+	// The capture classes of the ship being boarded. Stored in a list paired by
+	// the capture class name and the capture class itself. The list is sorted by
+	// an increasing success chance from the capture class.
+	std::vector<std::pair<std::string, CaptureClass>> captureRequirements;
```

Comment:
These two feel a bit huge and use a pair + vector and map so we probably should make a small class combining it?
~~I thought we were discouraging the usage of pair to begin with given it contains unnamed data~~

We also have to explain a lot when a good  + functions class name would do that everywhere we use it, I see that there are some differences for sure though so maybe its not easy because of that, but maybe logic could be moved to it, too?

Maybe its fine because the pair contains the name of the thing that is stored with it?